contracts/cw-proposal-single/src/tests.rs

```diff
@@ -4166,3 +4166,361 @@ fn test_return_deposit_to_dao_on_proposal_failure() {
     // Deposit should now belong to the DAO.
     assert_eq!(balance.balance, Uint128::new(1));
 }
+
+#[test]
+fn test_find_proposals() {
```

Comment:
Love how comprehensive this test is. :)

Reply:
Was it a hint that I should separate pagination into a separate test? Or we ok with this? :DD

Reply:
Aha - is a genuine compliment.

Reply:
Still added extra blocks and comments for easier read (: